tls/s2n_record_write.c

```diff
@@ -408,8 +408,9 @@ int s2n_record_writev(struct s2n_connection *conn, uint8_t content_type, const s
         conn->client = current_client_crypto;
         conn->server = current_server_crypto;
     }
-
+    conn->encrypted_bytes_out += encrypted_length;
```

Comment:
"There are cryptographic limits on the amount of plaintext which can
   be safely encrypted under a given set of keys" 

It seems that the limits from the RFC was mean't to be check against plaintext size, however since encrypted length is greater than plaintext length, this should not an issue.